reference/client/get-pages.mdx

```diff
@@ -0,0 +1,366 @@
+have a large number of pages, you may need to paginate through the results to
+retrieve all of them.
+
+To make this easier, we've implemented the return type of `getPages` as an
+extension of a `Promise` and an async iterator. This abstraction enables you to
+consume the pages in multiple ways, enabling you to easily retrieve all pages
+without worrying about the implementation details of pagination:
+
+```ts get-pages.ts
+import { MakeswiftPage } from "@makeswift/runtime/next"
+import { client } from "makeswift/client"
+
+export async function getAllPages() {
+  // The `toArray` method on the iterator will return all pages as an array
+  const pagesFromToArray = await client.getPages().toArray()
+
+  // You can also get all pages with a `for await` loop
+  const pagesFromForAwait: MakeswiftPage[] = []
+  for await (const page of client.getPages()) {
+    pages.push(page)
+  }
+
+  // Get a single paginated result set 
+  const { data: pagesFromPagination } = await client.getPages({ limit: 5 })
+}
+```
+
+The custom async iterator returned by `getPages` includes the following methods:
+
+- `toArray()`: Consumes all pages in the iterator and returns them as an array.
+
+- `map(callback: (page: MakeswiftPage) => T)`: Maps over each page in the iterator to yield a new data
+  type. Returns another async iterator.
+
+- `filter(predicate: () => Boolean)`: Selectively yields pages based on a
+  predicate function. Returns another async iterator.
+
+Because `.map` and `.filter` return another async iterator, you can chain these
+methods together:
+
+```ts
+const pages = await client.getPages()
+  .filter((page) => !page.excludedFromSearch)
+  .map((page) => ({ id: page.id, title: page.title }))
+  .toArray()
+```
+
+For more details on async iterators, see the [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/AsyncGenerator).
+
+The pagination for this method is cursor based, meaning that you can pass an ID
+of a page to the `after` option to retrieve the next set of pages after that
+page. This can be used to manually implement paginating through all pages:
+
+```ts get-pages.ts
+import { MakeswiftPage } from "@makeswift/runtime/next"
+import { client } from "makeswift/client"
+
+export async function getAllPages() {
+  const pages: MakeswiftPages[] = []
+  let cursor: string | undefined = undefined
+  let hasMorePages = true
+
+  do {
+    const {data, hasMore} = await client.getPages({ limit: 5, after: cursor })
+    pages.push(...paginatedResults.data)
+    hasMorePages = paginatedResults.hasMore
+    cursor = paginatedResults.data.at(-1)?.id
+  } while (hasMorePages)
+
+  return pages
+}
+```
+
+### Site Versions
+
+By default, `getPages` will return data reflecting the state of published
+Makeswift pages. If you need to retrieve draft information about pages as they
+currently are in the Makeswift builder, you can pass `Working` to the
```

Comment:
Up to this point we've treated `siteVersion` as an opaque value that you need to get from the `Makeswif.getSiteVersion` function or the `getSiteVersion` function in App Router.

At this point it's not a big deal to point out `'Working'` and `'Live'` are the values, but I just wanted to point that out. The reason is that we know we're moving towards branching and merging and we were going to potentially have other values here. At this point, though, we'll probably have a breaking change anyway once we release the changes.